app/routes/tasks.js

```diff
@@ -5,11 +5,20 @@ const API_BASE_URL = ENV.BASE_API_URL;
 
 export default class TasksRoute extends Route {
   model = async () => {
-    const response = await fetch(`${API_BASE_URL}/tasks/self`, {
-      credentials: 'include',
-    });
-    const allTasks = await response.json();
-
-    return allTasks;
+    try {
+      const response = await fetch(`${API_BASE_URL}/tasks/self`, {
+        credentials: 'include',
+      });
+      const allTasks = await response.json();
```

Comment:
I feel this statement could be at the bottom (after the checks) so that if the response is 401, we don't `wait` here converting the response to json. What do you think?

Reply:
Makes sense, Thanks for pointing out! Also I think `await` is not required in line 12 as we have already used it in line 9? :thinking:

Reply:
Check the return value here: https://developer.mozilla.org/en-US/docs/Web/API/Response/json
Can you tell me now why should we use `await` and if not why should we not use `await` 🙂

Reply:
Ohhh, Got it! We'll have use await here because response is not the actual response but the promise to return the response

Reply:
`The json() method of the Response interface takes a Response stream and reads it to completion. It returns a promise which resolves with the result of parsing the body text as JSON.`

Since it returns a promise, we want the promise to be resolved and that's why `await`

Whatever that returns a promise and we want to wait till it get resolved, we want to `await ` there

Hope that makes sense. This is going to be super super super important as you work on more features and projects. Promises are really important

Reply:
Got it. Thank you!